### PR TITLE
Added shutdown_timeout as part of execd

### PR DESCRIPTION
**Pull request by shriyanshk128T**
### Required for all PRs:

- [x] Signed [CLA](https://influxdata.com/community/cla/).
- [x] Associated README.md updated.
- [x] Has appropriate unit tests.

### CODE REVIEW

File: plugins/outputs/execd/execd.go [hunk 1]
```diff
@@ -31,6 +34,7 @@ const sampleConfig = `
 type Execd struct {
 	Command      []string        `toml:"command"`
 	RestartDelay config.Duration `toml:"restart_delay"`
+	Timeout      time.Duration   `toml:"timeout"`
```
Comment (BenMatase) [inline plugins/outputs/execd/execd.go]: Can you make the default 5s?

File: plugins/outputs/execd/execd.go [hunk 2]
```diff
@@ -21,6 +21,9 @@ const sampleConfig = `
   ## Delay before the process is restarted after an unexpected termination
   restart_delay = "10s"
 
+  ## Timeout 
+  timeout = 0
```
Comment (BenMatase) [inline plugins/outputs/execd/execd.go]: And then make the sample configs `timeout = 5s`

File: plugins/inputs/execd/README.md [hunk 3]
```diff
@@ -34,6 +34,9 @@ STDERR from the process will be relayed to Telegraf as errors in the logs.
   ## Delay before the process is restarted after an unexpected termination
   restart_delay = "10s"
 
+  ## Timeout
```
Comment (BenMatase) [inline plugins/inputs/execd/README.md]: Make a bit more descriptive. Something like `Timeout given to process when gracefully shutting down`

**Review by BenMatase — CHANGES_REQUESTED** [review]

File: plugins/inputs/execd/README.md [hunk 4]
```diff
@@ -34,6 +34,9 @@ STDERR from the process will be relayed to Telegraf as errors in the logs.
   ## Delay before the process is restarted after an unexpected termination
   restart_delay = "10s"
 
+  ## Timeout 
+  timeout = 0
```
Comment (BenMatase) [inline plugins/inputs/execd/README.md]: Can we make this `shutdown_timeout`?

File: internal/process/process.go [hunk 5]
```diff
@@ -169,7 +170,7 @@ func (p *Process) cmdWait(ctx context.Context) error {
 	go func() {
 		select {
 		case <-ctx.Done():
-			gracefulStop(processCtx, p.Cmd, 5*time.Second)
+			gracefulStop(processCtx, p.Cmd, shutdownTimeout*time.Second)
```
Comment (BenMatase) [inline internal/process/process.go]: You don't need the *time Second. In fact it might make it incorrect

File: plugins/inputs/execd/execd.go [hunk 6]
```diff
@@ -171,8 +176,9 @@ func (e *Execd) Init() error {
 func init() {
 	inputs.Add("execd", func() telegraf.Input {
 		return &Execd{
-			Signal:       "none",
-			RestartDelay: config.Duration(10 * time.Second),
+			Signal:          "none",
+			RestartDelay:    config.Duration(10 * time.Second),
+			ShutdownTimeout: 5,
```
Comment (BenMatase) [inline plugins/inputs/execd/execd.go]: Specify the *time.Second here

File: plugins/inputs/execd/execd_test.go [hunk 7]
```diff
@@ -26,6 +26,7 @@ func TestSettingConfigWorks(t *testing.T) {
 	[[inputs.execd]]
 		command = ["a", "b", "c"]
 		restart_delay = "1m"
+		shutdown_timeout = 5
```
Comment (BenMatase) [inline plugins/inputs/execd/execd_test.go]: This should be "5s" in all of the tomls

**Review by agrawalkaushik — APPROVED** [review]

File: plugins/inputs/execd/execd_test.go [hunk 8]
```diff
@@ -26,6 +26,7 @@ func TestSettingConfigWorks(t *testing.T) {
 	[[inputs.execd]]
 		command = ["a", "b", "c"]
 		restart_delay = "1m"
+		shutdown_timeout = "5s"
```
Comment (BenMatase) [inline plugins/inputs/execd/execd_test.go:29]: nit: can you do a value that isn't the default to make sure it isn't just using the default?

**Review by BenMatase — APPROVED** [review]

**Review by gregschrock — APPROVED** [review]

**Comment by gregschrock** [comment]
Please use "Squash" when merging the PR

**Pull request merged**